route/ipv6seed.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019 The CovenantSQL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package route
+
+import (
+	"fmt"
+
+	"github.com/CovenantSQL/beacon/ipv6"
+	"github.com/pkg/errors"
+
+	"github.com/CovenantSQL/CovenantSQL/crypto"
+	"github.com/CovenantSQL/CovenantSQL/crypto/asymmetric"
+	"github.com/CovenantSQL/CovenantSQL/pow/cpuminer"
+	"github.com/CovenantSQL/CovenantSQL/proto"
+)
+
+const (
+	NODEID = "id."
+	PUBKEY = "pub."
+	NONCE  = "n."
+	ADDR   = "addr."
+)
+
+type IPv6SeedClient struct{}
+
+func (isc *IPv6SeedClient) GetBPFromDNSSeed(BPDomain string) (BPNodes IDNodeMap, err error) {
+	// Public key
+	pubKeyBuf := make([]byte, asymmetric.PublicKeyBytesLen)
+	pubKeyBuf[0] = asymmetric.PublicKeyFormatHeader
+	var pubBuf, nonceBuf, addrBuf []byte
+	if pubBuf, err = ipv6.FromDomain(PUBKEY + BPDomain); err != nil {
+		return
+	}
+	if len(pubBuf) != asymmetric.PublicKeyBytesLen-1 {
+		return nil, errors.Errorf("error public key bytes len: %d", len(pubBuf))
+	}
+	copy(pubKeyBuf[1:], pubBuf)
+	var pubKey asymmetric.PublicKey
+	err = pubKey.UnmarshalBinary(pubKeyBuf)
+	if err != nil {
+		return
+	}
+
+	// Nonce
+	if nonceBuf, err = ipv6.FromDomain(NONCE + BPDomain); err != nil {
+		return
+	}
+	nonce, err := cpuminer.Uint256FromBytes(nonceBuf)
+	if err != nil {
+		return
+	}
+
+	// Addr
+	addrBuf, err = ipv6.FromDomain(ADDR + BPDomain)
+	if err != nil {
+		return
+	}
+	addrBytes, err := crypto.RemovePKCSPadding(addrBuf)
+	if err != nil {
+		return
+	}
+
+	// NodeID
+	nodeIdBuf, err := ipv6.FromDomain(NODEID + BPDomain)
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
var nodeIdBuf should be nodeIDBuf